R/hdf5.R

```diff
@@ -68,226 +72,133 @@ validHdf5MSnExp <- function(object) {
           to
       }
 
-serialise_to_hdf5 <- function(object, filename = NULL) {
-    stopifnot(inherits(object, "OnDiskMSnExp"))
-    if (is.null(filename))
-        filename <- paste0(digest::sha1(fileNames(object)), ".h5")
-    if (file.exists(filename))
-        stop("File ", filename, " already exists.")
-    h5 <- rhdf5::H5Fcreate(filename)
-    pb <- progress::progress_bar$new(total = length(object))
+#' Write the content of a single mzML/etc file to an h5file. We're using the
+#' spectrum index in the file as data set ID.
+#'
+#' @author Johannes Rainer
+#'
+#' @noRd
+.serialize_msfile_to_hdf5 <- function(file, h5file) {
+    h5 <- rhdf5::H5Fcreate(h5file)
     comp_level <- .hdf5_compression_level()
-    for (i in seq_along(fileNames(object))) {
-        file_name  <- fileNames(object)[i]
-        file_group <- .hdf5_group_name(file_name)
-        stopifnot(rhdf5::h5createGroup(h5, file_group))
-        fns <- featureNames(filterFile(object, i))
-        fh <- openMSfile(file_name)
-        pks <- peaks(fh)
-        close(fh)
-        for (j in seq_along(fns)) {
-            pb$tick()
-            fn <- fns[j]
-            hdfile <- paste0(file_group, "/", fn)
-            .pks <- pks[[j]]
-            colnames(.pks) <- c("mz", "intensity")
-            rhdf5::h5write(.pks, h5, hdfile, level = comp_level)
-        }
+    fh <- openMSfile(file)
+    hdr <- header(fh)
```

Comment:
In case you're wondering why we read also the `header` although we don't *need* that data: without the `header` call I got randomly a ProteoWizard error that the lengths of mz and intensity values for a spectrum don't match.